### DISCUSSION TITLE
Use preinstalled programs in vcpkg_find_acquire_program()?

### DISCUSSION THREAD

**Discussion by KUGA2** [microsoft/vcpkg#34376] [General]
Hi,

Is it possible to use preinstalled programs in [vcpkg_find_acquire_program()](https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_find_acquire_program)?

My specific example: port 'openssl' for Windows needs NASM, which it always downloads. Even when I installed it via chocolatey (To `C:\Program Files\NASM`). But from what I have learned by reading the code this is true for every program!

How to find the installed program: Add the path to [paths_to_search](https://github.com/microsoft/vcpkg/blob/cc97b4536ae749ec0e4f643488b600b217540fb3/scripts/cmake/vcpkg_find_acquire_program(NASM).cmake#L3)

Is there any way without patching vcpkg?

I could add `C:\Program Files\NASM` to path and run with VCPKG_ENV_PASSTHROUGH=PATH (i guess), but then [`NO_DEFAULT_PATH  `](https://github.com/microsoft/vcpkg/blob/cc97b4536ae749ec0e4f643488b600b217540fb3/scripts/cmake/vcpkg_find_acquire_program.cmake#L79) would block that. Why did you add `NO_DEFAULT_PATH` there in the first place?

btw: Why does [`VCPKG_FORCE_SYSTEM_BINARIES`](https://learn.microsoft.com/en-us/vcpkg/users/config-environment#vcpkg_force_system_binaries) have no effect on this at all?

**Comment by autoantwort** [comment]
> Why does [VCPKG_FORCE_SYSTEM_BINARIES](https://learn.microsoft.com/en-us/vcpkg/users/config-environment#vcpkg_force_system_binaries) have no effect on this at all?

Because it only affect tools that are downloaded by vcpkg-tool from the vcpkgTools.xml file.

**Comment by dg0yt** [comment]
> I could add `C:\Program Files\NASM` to path and run with VCPKG_ENV_PASSTHROUGH=PATH (i guess), but then [`NO_DEFAULT_PATH `](https://github.com/microsoft/vcpkg/blob/cc97b4536ae749ec0e4f643488b600b217540fb3/scripts/cmake/vcpkg_find_acquire_program.cmake#L79) would block that. Why did you add `NO_DEFAULT_PATH` there in the first place?

Don't guess. Do it.

This `NO_DEFAULT_PATH` is used in the function which explicitly checks for vcpkg's downloaded binary. It doesn't block the detecting the external binary:
https://github.com/microsoft/vcpkg/blob/cc97b4536ae749ec0e4f643488b600b217540fb3/scripts/cmake/vcpkg_find_acquire_program.cmake#L40

**Reply by KUGA2** [reply]
You are correct. It works. Even in my old version of vcpkg.
